### ISSUE TITLE
Using Orca Estimator get_model method cannot return keras model object.

### ISSUE THREAD

**Issue by GitEasonXu** [intel-analytics/analytics-zoo#3401]
Using Orca Estimator get_model method cannot return keras model object. Maybe it is related to `BN` layer.
```python
est.save("/tmp/yolov1.ckpt")
est.restore("/tmp/yolov1.ckpt")
model = est.get_model()
model.save("/tmp/yolov1.h5")
```

**Comment by GitEasonXu** [comment]
```python
    def get_model(self):
        """Returns the learned model."""
        state = ray.get(self.remote_workers[0].get_state.remote())
        #state_refs = [w.get_state.remote() for w in self.remote_workers]
        #state = ray.get(state_refs[0])
        
        return self._get_model_from_state(state)
```
Should I change `state = ray.get(self.remote_workers[0].get_state.remote())` to `state_refs = [w.get_state.remote() for w in self.remote_workers]  state = ray.get(state_refs[0])`

**Comment by yangw1234** [comment]
> ```python
>     def get_model(self):
>         """Returns the learned model."""
>         state = ray.get(self.remote_workers[0].get_state.remote())
>         #state_refs = [w.get_state.remote() for w in self.remote_workers]
>         #state = ray.get(state_refs[0])
>         
>         return self._get_model_from_state(state)
> ```
> 
> Should I change `state = ray.get(self.remote_workers[0].get_state.remote())` to `state_refs = [w.get_state.remote() for w in self.remote_workers] state = ray.get(state_refs[0])`
Yeah, I think that's the correct fix. Could you try if that works?

**Comment by GitEasonXu** [comment]
Need change the two methods, and I have tested it without problems.
```python     
def get_model(self):
        """Returns the learned model."""
        #state = ray.get(self.remote_workers[0].get_state.remote())

        state_refs = [w.get_state.remote() for w in self.remote_workers]
        state = ray.get(state_refs[0])

        return self._get_model_from_state(state)
```
```python
    def _get_model_from_state(self, state):
        """Creates model and load weights from state"""

        model = self.model_creator(self.config)
        model.set_weights(state["weights"])

        # This part is due to ray.get() changing scalar np.int64 object to int
        if state["optimizer_weights"]:
            state["optimizer_weights"] = np.array(
                state["optimizer_weights"], dtype=np.int64)

            if model.optimizer.weights == []:
                model.make_train_function()
            model.optimizer.set_weights(state["optimizer_weights"])
        
        return model
```

**Comment by GitEasonXu** [comment]
Update
```python
    def _get_model_from_state(self, state):
        """Creates model and load weights from state"""

        model = self.model_creator(self.config)
        model.set_weights(state["weights"])
        
        # This part is due to ray.get() changing scalar np.int64 object to int
        if state["optimizer_weights"]:
            state["optimizer_weights"][0] = np.array(
                state["optimizer_weights"][0], dtype=np.int64)
            if model.optimizer.weights == []:
                model.make_train_function()
                
            if len(model.optimizer.get_weights()) == len(state["optimizer_weights"]):
                model.optimizer.set_weights(state["optimizer_weights"])

        return model
```

**Comment by yangw1234** [comment]
great @GitEasonXu we'll submit a pr to fix this.

**Comment by leonardozcm** [comment]
> Update
> 
> ```python
>     def _get_model_from_state(self, state):
>         """Creates model and load weights from state"""
> 
>         model = self.model_creator(self.config)
>         model.set_weights(state["weights"])
>         
>         # This part is due to ray.get() changing scalar np.int64 object to int
>         if state["optimizer_weights"]:
>             state["optimizer_weights"][0] = np.array(
>                 state["optimizer_weights"][0], dtype=np.int64)
>             if model.optimizer.weights == []:
>                 model.make_train_function()
>                 
>             if len(model.optimizer.get_weights()) == len(state["optimizer_weights"]):
>                 model.optimizer.set_weights(state["optimizer_weights"])
> 
>         return model
> ```

hi @GitEasonXu ,I think  `model.optimizer.set_weights(state["optimizer_weights"])` will be skipped for `model.optimizer.get_weights()` is still empty after 'make_train_function()', would you mind showing us more info about error you met in `_get_model_from_state`?

**Comment by leonardozcm** [comment]
> > Update
> > ```python
> >     def _get_model_from_state(self, state):
> >         """Creates model and load weights from state"""
> > 
> >         model = self.model_creator(self.config)
> >         model.set_weights(state["weights"])
> >         
> >         # This part is due to ray.get() changing scalar np.int64 object to int
> >         if state["optimizer_weights"]:
> >             state["optimizer_weights"][0] = np.array(
> >                 state["optimizer_weights"][0], dtype=np.int64)
> >             if model.optimizer.weights == []:
> >                 model.make_train_function()
> >                 
> >             if len(model.optimizer.get_weights()) == len(state["optimizer_weights"]):
> >                 model.optimizer.set_weights(state["optimizer_weights"])
> > 
> >         return model
> > ```
> 
> hi @GitEasonXu ,I think `model.optimizer.set_weights(state["optimizer_weights"])` will be skipped for `model.optimizer.get_weights()` is still empty after `make_train_function()`, would you mind showing us more info about error you met in `_get_model_from_state`?

**Comment by leonardozcm** [comment]
Fixed at #3418
